contract_sale_invoicing/models/contract.py

```diff
@@ -0,0 +1,29 @@
+# Copyright 2018 Tecnativa - Carlos Dauden
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class AccountAnalyticAccount(models.Model):
+    _inherit = 'account.analytic.account'
+
+    invoicing_sales = fields.Boolean(
+        string='Invoicing Sales',
+        help='If checked to invoce sales will be invoiced on invoice '
+             'recurring invoice creation'
+    )
+
+    @api.multi
+    def _create_invoice(self, invoice=False):
+        if not self.invoicing_sales:
+            return super(AccountAnalyticAccount, self)._create_invoice()
+        sales = self.env['sale.order'].search([
+            ('partner_invoice_id', '=', self.partner_id.id),
+            ('invoice_status', '=', 'to invoice'),
+            ('date_order', '<=',
+             '{} 23:59:59'.format(self.recurring_next_date)),
```

Comment:
Shouldn't this use the end date determined by the contract? (for taking into account pre-paid/post-paid)

Reply:
This is the date that the invoice will have, so for me it is the logical limit.
If someone prefers another limit he just has to extend the function

Reply:
OK